parser_library/src/context/instruction.cpp

```diff
@@ -445,1252 +443,1396 @@ hlasm_plugin::parser_library::context::instruction::get_machine_instructions()
         { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U },
-        48,
+
         1569);
     add_machine_instr(
-        result, "VMAO", mach_format::VRR_d, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 48, 1570);
+        result, "VMAO", mach_format::VRR_d, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 1570);
     add_machine_instr(result,
         "VMALO",
         mach_format::VRR_d,
         { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U },
-        48,
+
         1570);
-    add_machine_instr(result, "VMH", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 48, 1570);
-    add_machine_instr(
-        result, "VMLH", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 48, 1571);
-    add_machine_instr(result, "VML", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 48, 1571);
-    add_machine_instr(result, "VME", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 48, 1572);
-    add_machine_instr(
-        result, "VMLE", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 48, 1572);
-    add_machine_instr(result, "VMO", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 48, 1572);
-    add_machine_instr(
-        result, "VMLO", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 48, 1572);
+    add_machine_instr(result, "VMH", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 1570);
+    add_machine_instr(result, "VMLH", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 1571);
+    add_machine_instr(result, "VML", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 1571);
+    add_machine_instr(result, "VME", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 1572);
+    add_machine_instr(result, "VMLE", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 1572);
+    add_machine_instr(result, "VMO", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 1572);
+    add_machine_instr(result, "VMLO", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U }, 1572);
     add_machine_instr(result,
         "VMSL",
         mach_format::VRR_d,
         { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, vec_reg_4_U, mask_4_U, mask_4_U },
-        48,
+
         1573);
-    add_machine_instr(result, "VNN", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U }, 48, 1574);
-    add_machine_instr(result, "VNO", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U }, 48, 1574);
+    add_machine_instr(result, "VNN", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U }, 1574);
+    add_machine_instr(result, "VNO", mach_format::VRR_c, { vec_reg_4_U, vec_reg_4_U, vec_reg_4_U }, 1574);
     result.insert(std::pair<const std::string, machine_instruction_ptr>("VNOT",
```

Comment:
I am wondering about this mnemonic. As far as I can tell VNOT has two operands VNOT A,B which is expanded into VNO A,B,B instruction. Yet the plugin requires three arguments at the moment.

Reply:
Yes, you are correct. But , this was something pre-existing in the plugin where it takes 3 arguments and wants 2nd and 3rd operands to be equal. I am not sure why it was done this way ?
Maybe @michalbali256 remembers something about it

Reply:
So VNOT is actually a mnemonic? It must have been originally misunderstood... We can either keep VNOT as an instruction and just edit its format, or create a mnemonic - slight problem being, that we do not have a way to copy second operand and use it as third. However, we can just substitute third operand with zero, which will always be valid. At the same time, the second operand will still be checked so the user will not know.

Either way, class `vnot_instruction` can be removed.